tools/gen-intrinsics-specs.py

```diff
@@ -93,51 +94,83 @@ def rst_literal_quote(mapping, workflow):
         return '```\n' + '\n'.join(lines) + '\n```'
 
 
-def quote_split_intrinsics(intrinsic, workflow):
+def quote_split_intrinsics(intrinsic, workflow, baseurl=__ARMDEVELOPER):
     r"""
     >>> quote_split_intrinsics('int f(int x, float y)', 'rst')
     '.. code:: c\n\n    int f(\n        int x,\n        float y)'
 
     >>> quote_split_intrinsics('int f(int x)', 'rst')
     '.. code:: c\n\n    int f(int x)\n'
 
-    >>> quote_split_intrinsics('int f(int x, float y)', 'markdown')
-    '`int f(`<br>&nbsp;&nbsp;&nbsp;&nbsp;`int x,`<br>&nbsp;&nbsp;&nbsp;&nbsp;` float y)`'
+    >>> quote_split_intrinsics('int f(int x, float y)', 'markdown', 'baseurl/')
+    '<code>int <a href="baseurl/f" target="_blank">f</a>(<br>&nbsp;&nbsp;&nbsp;&nbsp; int x,<br>&nbsp;&nbsp;&nbsp;&nbsp; float y)</code>'
 
-    >>> quote_split_intrinsics('int f(int x)', 'markdown')
-    '`int f(int x)`'
+    >>> quote_split_intrinsics('int f(int x)', 'markdown', 'baseurl/')
+    '<code>int <a href="baseurl/f" target="_blank">f</a>(int x)</code>'
 
-    >>> quote_split_intrinsics('int f(int x, float y)', 'pdf')
+    >>> quote_split_intrinsics('int f(int x, float y)', 'pdf', 'baseurl/')
     '``` c\nint f(\n  int x,\n  float y)\n```'
 
-    >>> quote_split_intrinsics('int f(int x)', 'pdf')
+    >>> quote_split_intrinsics('int f(int x)', 'pdf', 'baseurl/')
     '`int f(int x)`'
     """
     # Remove the suffix ')' from the intrinsic string.
     intrinsic_without_ending = intrinsic[:-1]
-    ret_def, par, signature = intrinsic_without_ending.partition('(')
-    split_signature = signature.split(',')
     whitespace_indent = "&nbsp;&nbsp;&nbsp;&nbsp;"
 
-    if len(split_signature) > 1:
+    intrinsic_signature = get_signature(intrinsic_without_ending)
+    intrinsic_link_id = get_intrinsic_name(intrinsic_without_ending)
+    intrinsic_type = get_intrinsic_return_type(intrinsic_without_ending)
```

Comment:
I think there is no need to compute `intrinsic_without_ending` as these functions are tested with the whole `intrinsic` string from the csv file.

Please change the input to be `intrinsic` and remove computation of `intrinsic_without_ending` in line 118.